src/paths/steiner.cpp

```diff
@@ -0,0 +1,252 @@
+	IGRAPH_FINALLY(igraph_matrix_destroy,&dp_cache);
+
+    igraph_matrix_fill(&dp_cache, (igraph_real_t) IGRAPH_INTEGER_MAX);
+
+//	printf("Matrix Filled\n");
+	
+	q = VECTOR(steiner_terminals_copy)[0];
+
+	igraph_vector_int_remove(&steiner_terminals_copy, 0);
+
+	allSubsets = generateSubsets(steiner_terminals_copy, igraph_vector_int_size(steiner_terminals), no_of_vertices);
+//	printSubsets(allSubsets);
+	// Singleton subset rows may be filled in trivially
+	// printf("subsets Filled\n");
+	// printf("Size:%ld\n",igraph_vector_int_size(&steiner_terminals_copy));
+	// for (igraph_integer_t i = 0; i < igraph_vector_int_size(&steiner_terminals_copy); i++)
+	// {
+	// 	printf("I am Looping n\n");
+	// 	printf("%ld",igraph_vector_size(&steiner_vertices));
+	// 	for (igraph_integer_t j = 0; j < igraph_vector_size(&steiner_vertices); j++)
+	// 	{
+	// 		printf("I am Looping mx n\n");
+	// 		printf("%f",MATRIX(distance, (igraph_integer_t)VECTOR(steiner_terminals_copy)[i], (igraph_integer_t)VECTOR(steiner_vertices)[j]));
+	// 		igraph_matrix_set(&dp_cache, (igraph_integer_t)VECTOR(steiner_terminals_copy)[i], (igraph_integer_t)VECTOR(steiner_vertices)[j], MATRIX(distance, (igraph_integer_t)VECTOR(steiner_terminals_copy)[i], (igraph_integer_t)VECTOR(steiner_vertices)[j]));
+	// 	}
+	// }
+	for (igraph_integer_t m = 2; m <= igraph_vector_int_size(&steiner_terminals_copy); m++)
+	{
+		for (igraph_integer_t i = 0; i < (igraph_integer_t)allSubsets.size(); i++)
+		{
+			auto it = allSubsets.begin();
+			std::advance(it,i);
+			std::set<igraph_integer_t> D = *it;
+			igraph_integer_t indexOfSubsetD;
+			indexOfSubsetD = fetchIndexofMapofSets(D);
+			for (igraph_integer_t j = 0; j < igraph_vector_size(&steiner_vertices); j++)
+			{
+				igraph_integer_t u = IGRAPH_INTEGER_MAX;
+				std::set<igraph_integer_t>::iterator subset_D_iterator;
+
+				for (subset_D_iterator = D.begin(); subset_D_iterator != D.end(); subset_D_iterator++)
+				{
+					igraph_integer_t E = *subset_D_iterator;
+
+					igraph_integer_t distanceEJ = MATRIX(distance, E, j);
+					std::set<igraph_integer_t> DMinusE = D;
+
+					//igraph_vector_remove(&DMinusE,E);
+
+					for (std::set<igraph_integer_t>::iterator iter = DMinusE.begin(); iter != DMinusE.end();)
+					{
+						if (*iter == E)
+						{
+							iter = DMinusE.erase(iter);
+						}
+						else
+						{
+							++iter;
+						}
+					}
+
+					igraph_integer_t indexOfSubsetDMinusE = fetchIndexofMapofSets(DMinusE);
+
+					if (distanceEJ + (MATRIX(dp_cache, indexOfSubsetDMinusE, j)) < u)
+					{
+						u = distanceEJ + (MATRIX(dp_cache, indexOfSubsetDMinusE, j));
+					}
+				}
+				for (igraph_integer_t k = 0; k < igraph_vector_size(&steiner_vertices); k++)
+				{
+					MATRIX(dp_cache, indexOfSubsetD, k) = std::min(MATRIX(dp_cache, indexOfSubsetD, k), MATRIX(distance, k, j) + u);
+				}
+			}
+		}
+	}
+	igraph_integer_t u = IGRAPH_INTEGER_MAX;
+	igraph_integer_t v = IGRAPH_INTEGER_MAX;
```

Comment:
Are these variables representing a distance? If so, can you please choose a different name for them? `u` and `v` are traditionally used to denote vertices (in igraph: vertiex IDs).

If these are distances, should they not be of type `igraph_real_t`? In the weighted case distances are not necessarily integer valued.

Reply:
Please click "Resolve conversation" under all review comments that have been addressed (unless there's anything left to do). This way we can keep track of what still needs work.